Source/WebKit/NetworkProcess/storage/CacheStorageDiskStore.cpp

```diff
@@ -506,8 +506,9 @@ void CacheStorageDiskStore::writeRecords(Vector<CacheStorageRecord>&& records, W
         auto bodyData = encodeRecordBody(record);
         auto bodyHash = computeSHA1(bodyData.span(), m_salt);
         bool shouldCreateBlob = shouldStoreBodyAsBlob(bodyData);
+        auto recordInfoKey = record.info.key;
         auto headerData = encodeRecordHeader(WTFMove(record));
-        auto recordData = encodeRecord(record.info.key, headerData, !shouldCreateBlob, bodyData, bodyHash, m_salt);
```

Comment:
Why do we need to pass a && to encodeRecordHeader, it seems we could pass a const&

Reply:
To move `record.responseData` and pass to ResourceResponse::fromCrossThreadData?

Reply:
Ah I see, maybe we should have an encoder that takes a cross thread data, this could be left to another PR though.